libresapi/src/api/TransfersHandler.cpp

```diff
@@ -6,20 +6,33 @@
 namespace resource_api
 {
 
-TransfersHandler::TransfersHandler(StateTokenServer *sts, RsFiles *files, RsPeers *peers):
-    mStateTokenServer(sts), mFiles(files), mRsPeers(peers), mLastUpdateTS(0)
+TransfersHandler::TransfersHandler(StateTokenServer *sts, RsFiles *files, RsPeers *peers,
+                                   RsNotify* notify):
+    mStateTokenServer(sts), mFiles(files), mRsPeers(peers), mLastUpdateTS(0), mNotify(notify)
 {
 	addResourceHandler("*", this, &TransfersHandler::handleWildcard);
 	addResourceHandler("downloads", this, &TransfersHandler::handleDownloads);
 	addResourceHandler("uploads", this, &TransfersHandler::handleUploads);
 	addResourceHandler("control_download", this, &TransfersHandler::handleControlDownload);
 	mStateToken = mStateTokenServer->getNewToken();
 	mStateTokenServer->registerTickClient(this);
+	mNotify->registerNotifyClient(this);
 }
 
 TransfersHandler::~TransfersHandler()
 {
     mStateTokenServer->unregisterTickClient(this);
+	mNotify->unregisterNotifyClient(this);
+}
+
+void TransfersHandler::notifyListChange(int list, int /* type */)
+{
+	//RsStackMutex stack(mMtx); /********** STACK LOCKED MTX ******/
```

Comment:
If the mutex locking is not needed this line of potential deadcode should be removed before merging